eng/pipelines/templates/jobs/archetype-sdk-tests.yml

```diff
@@ -53,6 +54,31 @@ jobs:
           packageType: sdk
           version: "$(DotNetCoreSDKVersion)"
 
+      # TODO: Remove when eng/New-TestResources.ps1 bootstraps the Az module.
+      - pwsh: Install-Module -Name Az -Scope CurrentUser -Force
+        displayName: "Install Azure PowerShell module"
+
+        # Chomp the string literal to more easily read and update the New-TestResources parameters.
+      - pwsh: >
+          $baseName = 't' + (New-Guid).ToString('n').Substring(0, 16);
+          eng/New-TestResources.ps1
+          -BaseName $baseName
+          -ServiceDirectory '${{ parameters.ServiceDirectory }}'
+          -TenantId "${env:AZURE_TENANT_ID}"
+          -ProvisionerApplicationId "${env:AZURE_CLIENT_ID}"
+          -ProvisionerApplicationSecret "${env:AZURE_CLIENT_SECRET}"
+          -TestApplicationId "${env:AZURE_CLIENT_ID}"
+          -TestApplicationSecret "${env:AZURE_CLIENT_SECRET}"
+          -DeleteAfterHours 24
+          -Force
+          -Verbose
+        displayName: "Create test resources"
+        env:
```

Comment:
Azure DevOps is very good about redacting secret strings in the logs. We've been using environment variables as a safety measure to ensure that secrets are never output in the logs when that can be prevented.

Reply:
I agree with @pakrym here we shouldn't even put them in the env if we don't need to. We should just pass them as parameters. DevOps blindly hides these secrets from logs by a simple find/replace so that shouldn't be an issue.

Reply:
We have to. See my comments in the script. If we don't pass issecret=true variables as env vars, they aren't accessible. See https://docs.microsoft.com/en-us/azure/devops/pipelines/process/variables?view=azure-devops&tabs=yaml%2Cbatch#secret-variables. And, right now, they have to be environment variables for existing test code to use them (the step below).

Reply:
But isn't the script setting up the correct environment variables? I was sort of expecting we would pass in the secrets this script needs as variables, secrets can still be passed a parameters we do it all the time, and then the script would setup whatever variables are relevant for the tests.